chain-factory/src/factory.rs

```diff
@@ -71,195 +38,90 @@ pub trait FactoryModule:
             .from_source(source_address)
             .code_metadata(metadata)
             .returns(ReturnsNewManagedAddress)
-            .sync_call();
-
-        let chain_id = self.generate_chain_id();
-        self.set_deployed_contract_to_storage(
-            &caller,
-            chain_id,
-            ScArray::ChainConfig,
-            &chain_config_address,
-        );
+            .sync_call()
     }
 
-    #[only_admin]
+    #[only_owner]
     #[endpoint(deployHeaderVerifier)]
-    fn deploy_header_verifier(&self, bls_pub_keys: MultiValueEncoded<ManagedBuffer>) {
-        let caller = self.blockchain().get_caller();
-
-        require!(
-            self.is_chain_deployed(&caller),
-            "The current caller has not deployed and Sovereign Chain"
-        );
-
-        let chain_id = self.all_deployed_contracts(&caller).get().chain_id;
+    fn deploy_header_verifier(
+        &self,
+        bls_pub_keys: MultiValueEncoded<ManagedBuffer>,
+    ) -> ManagedAddress {
         let source_address = self.header_verifier_template().get();
-
-        self.require_bls_keys_in_range(&caller, &chain_id, bls_pub_keys.len().into());
-
         let metadata = self.blockchain().get_code_metadata(&source_address);
-        let header_verifier_address = self
-            .tx()
+
+        self.tx()
             .typed(HeaderverifierProxy)
             .init(bls_pub_keys)
             .gas(60_000_000)
             .from_source(source_address)
             .code_metadata(metadata)
             .returns(ReturnsNewManagedAddress)
-            .sync_call();
-
-        self.set_deployed_contract_to_storage(
-            &caller,
-            chain_id,
-            ScArray::SovereignHeaderVerifier,
-            &header_verifier_address,
-        );
+            .sync_call()
     }
 
-    #[only_admin]
-    #[endpoint(deployCrossChainOperation)]
-    fn deploy_cross_chain_operation(
+    #[only_owner]
+    #[endpoint(deployEnshrineEsdtSafe)]
+    fn deploy_enshrine_esdt_safe(
         &self,
         is_sovereign_chain: bool,
+        token_handler_address: ManagedAddress,
         opt_wegld_identifier: Option<TokenIdentifier>,
         opt_sov_token_prefix: Option<ManagedBuffer>,
```

Comment:
these should not be optional for main chain deploy
```suggestion
        wegld_identifier: TokenIdentifier,
        sov_token_prefix: ManagedBuffer,
```